### PR TITLE
proto: add machine_types and num_gpus option

### PR DESCRIPTION
**Pull request by chamini2**
closes FEA-3123

### CODE REVIEW

**Review by efiop — APPROVED** [review]

File: projects/isolate_proto/src/isolate_proto/controller_pb2_grpc.py [hunk 1]
```diff
@@ -1,34 +1,9 @@
...
-    warnings.warn(
-        f'The grpc package installed is at version {GRPC_VERSION},'
-        + f' but the generated code in controller_pb2_grpc.py depends on'
-        + f' grpcio>={GRPC_GENERATED_VERSION}.'
-        + f' Please upgrade your grpc module to grpcio>={GRPC_GENERATED_VERSION}'
-        + f' or downgrade your generated code using grpcio-tools<={GRPC_VERSION}.'
-        + f' This warning will become an error in {EXPECTED_ERROR_RELEASE},'
-        + f' scheduled for release on {SCHEDULED_RELEASE_DATE}.',
-        RuntimeWarning
-    )
```
Comment (efiop) [inline projects/isolate_proto/src/isolate_proto/controller_pb2_grpc.py:30]: @chamini2 Check that you've used correct grpc version to generate these. The fact that this is deleted hints me that you have something old installed.

**Review by efiop — CHANGES_REQUESTED** [review]
See comment about grpcio version

File: projects/isolate_proto/src/isolate_proto/controller_pb2_grpc.py (hunk 1 above)
Comment (chamini2) [reply projects/isolate_proto/src/isolate_proto/controller_pb2_grpc.py:30]: what version should i use

Comment (efiop) [reply projects/isolate_proto/src/isolate_proto/controller_pb2_grpc.py:30]: 1.64

https://github.com/fal-ai/fal/blob/main/projects/isolate_proto/pyproject.toml#L39

File: projects/isolate_proto/src/isolate_proto/controller_pb2.pyi [hunk 2]
```diff
@@ -878,11 +874,11 @@ class ListAliasRunnersResponse(google.protobuf.message.Message):
         *,
         runners: collections.abc.Iterable[global___RunnerInfo] | None = ...,
     ) -> None: ...
-    def ClearField(self, field_name: typing.Literal["runners", b"runners"]) -> None: ...
+    def ClearField(self, field_name: typing_extensions.Literal["runners", b"runners"]) -> None: ...
 
 global___ListAliasRunnersResponse = ListAliasRunnersResponse
 
-@typing.final
+@typing_extensions.final
```
Comment (chamini2) [inline projects/isolate_proto/src/isolate_proto/controller_pb2.pyi:881]: no idea about typing extensions vs typing

**Review by efiop — APPROVED** [review]
Feel free to merge when ready.

**Pull request merged**